File: yumyum-back/src/main/java/com/yumyum/domain/feed/dto/CreateFeedRequest.java
package com.yumyum.domain.feed.dto;

import com.yumyum.domain.feed.entity.Feed;
import com.yumyum.domain.map.dto.PlaceRequest;
import com.yumyum.domain.map.entity.Place;
import com.yumyum.domain.user.entity.User;
import lombok.*;

@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CreateFeedRequest {

    private String title;

    private String content;

    private Long score;

    private Long userId;

    private PlaceRequest placeRequest;

    private String videoPath;

    private String thumbnailPath;

    private Boolean isCompleted;

    public Feed toEntity(final User user, final Place place){
        return Feed.builder()
                .title(title)
                .score(score)
                .content(content)
                .user(user)
                .place(place)
                .videoPath(videoPath)
                .thumbnailPath(thumbnailPath)
                .isCompleted(isCompleted)
                .build();
    }
}

File: AOS/app/src/main/java/com/omnyom/yumyum/model/feed/CreateFeedRequest.kt
package com.omnyom.yumyum.model.feed

import java.io.Serializable


class CreateFeedRequest(
        private val content: String?,
        private val isCompleted: Boolean,
        private val placeRequest: PlaceRequest?,
        private val score : Int?,
        private val thumbnailPath : String?,
        private val title : String?,
        private val userId  : Long,
        private val videoPath : String

) {
    fun get() : HashMap<String, Any?> {
        return hashMapOf(
                "content" to content,
                "isCompleted" to isCompleted,
                "placeRequest" to placeRequest,
                "score" to score,
                "thumbnailPath" to thumbnailPath,
                "title" to title,
                "userId" to userId,
                "videoPath" to videoPath,
        )
    }
}

data class      PlaceRequest(
        var address: String,
        var locationX: Double,
        var locationY: Double,
        var name: String,
        var phone: String
) : Serializable
